src/types.tsx

```diff
@@ -242,21 +242,21 @@ export interface FormikRegistration {
 export type FormikContext<Values> = FormikProps<Values> &
   Pick<FormikConfig<Values>, 'validate' | 'validationSchema'>;
 
-export interface SharedRenderProps<T> {
+export interface SharedRenderProps<P> {
   /**
    * Field component to render. Can either be a string like 'select' or a component.
    */
-  component?: string | React.ComponentType<T | void>;
```

Comment:
Are there any downside/upsides of removing default `void` type?

Reply:
The use of `void` is what was causing the bug I was running into. When I was looking around to see how I don't see how types were passed to `React.ComponentType`, I didn't see any using `void` which is what lead to me try removing it. When removed it, I no longer had any errors.

Reply:
Also when I opened up `src/types.d.ts`, the following typescript errors were shown:

![image](https://user-images.githubusercontent.com/28444/54254207-2ee40700-4510-11e9-8061-d5da3ab1f868.png)

Again, removing the `| void` also resolved all these errors.

Reply:
Here's the actual `ComponentType` type from DefinitelyTyped:

```ts
type ComponentType<P = {}> = ComponentClass<P> | FunctionComponent<P>;
```
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L63

Note that `P` defaults to `{}` if unspecified… and thus `void` is not used AFAIK.